repo/db/cases.go

```diff
@@ -12,8 +12,11 @@ import (
 )
 
 type CasesDB struct {
-	db   *sql.DB
-	lock *sync.Mutex
+	modelStore
+}
+
+func NewCaseStore(db *sql.DB, lock *sync.Mutex) repo.CaseStore {
+	return &CasesDB{modelStore{db, lock}}
```

Comment:
`modelStore` acts as a base struct which we can capture all of the database-specific internals